src/main/java/frc/robot/Arm.java

```diff
@@ -16,44 +18,66 @@ public class Arm extends SubsystemBase{
-        motorTwo = new CANSparkMax(Constants.Arm.MOTOR_2_ID, com.revrobotics.CANSparkLowLevel.MotorType.kBrushless);
+        armEncoder = new Encoder(Constants.Arm.ENCODER_CHANNEL_A_ID, Constants.Arm.ENCODER_CHANNEL_B_ID);
+        motorOne = new CANSparkMax(Constants.Arm.ARM_MOTOR_ONE, MotorType.kBrushless);
+        motorTwo = new CANSparkMax(Constants.Arm.ARM_MOTOR_TWO, MotorType.kBrushless);
+        if (Robot.isSimulation()) {
+            REVPhysicsSim.getInstance().addSparkMax(motorOne, DCMotor.getNEO(1));
+            REVPhysicsSim.getInstance().addSparkMax(motorTwo, DCMotor.getNEO(1));
+        }
     }
 
-    public double getAngle(){
-        return (armEncoder.get()/Constants.Arm.ENCODER_COUNT)*360;
+    public double getAngle() {
+        return ((double) armEncoder.get() / Constants.Arm.ENCODER_COUNT) * 360;
     }
 
-    public boolean isOnTarget(){
-        return((targetAngle - Constants.Arm.TARGET_RANGE <= getAngle()) && (targetAngle + Constants.Arm.TARGET_RANGE >= getAngle()));
+    public boolean isOnTarget() {
+        return ((targetAngle - Constants.Arm.TARGET_RANGE_DEGREES <= getAngle())
+                && (targetAngle + Constants.Arm.TARGET_RANGE_DEGREES >= getAngle()));
     }
 
-    public double getTarget(){
+    public double getTarget() {
         return targetAngle;
     }
 
-    public void setTarget(double inputTarget){
+    public void setTarget(double inputTarget) {
         targetAngle = inputTarget;
     }
+
     public boolean isAtHomePostion() {
-        return((Constants.Arm.HOME_POSITION - Constants.Arm.HOME_POSITION_RANGE <= getAngle()) && (Constants.Arm.HOME_POSITION + Constants.Arm.HOME_POSITION_RANGE >= getAngle()));
+        if (homeSwitch.get()) {
+            return true;
+        } else {
+            return ((Constants.Arm.HOME_POSITION - Constants.Arm.HOME_POSITION_RANGE_DEGREES <= getAngle())
+                    && (Constants.Arm.HOME_POSITION + Constants.Arm.HOME_POSITION_RANGE_DEGREES >= getAngle()));
+
+        }
     }
 
     @Override
-    public void periodic(){
-        double motorSpeed = pid.calculate(targetAngle - getAngle());
-        if (homeSwitch.get()){
-            if (motorSpeed >= 0){
-                motorOne.set(motorSpeed);
-                motorTwo.set(motorSpeed);
+    public void periodic() {
+        double motorVoltage = pid.calculate(targetAngle - getAngle());
+        if (homeSwitch.get()) {
+            if (getAngle() >= Constants.Arm.HOME_SWITCH_FAILSAFE) {
+                motorOne.setVoltage(0);
+                motorTwo.setVoltage(0);
+            } else {
+                if (motorVoltage >= 0) {
+                    motorOne.setVoltage(motorVoltage);
+                    motorTwo.setVoltage(motorVoltage);
+                }
+            }
+        } else if (getAngle() == Constants.Arm.ARM_ANGLE_LIMIT_DEGREES) {
+            if (motorVoltage >= 0) {
+                motorOne.setVoltage(motorVoltage);
+                motorTwo.setVoltage(motorVoltage);
```

Comment:
This code block is almost correct, but one of these should set the voltage to zero.